server/conn.go

```diff
@@ -798,9 +798,46 @@ func (cc *clientConn) handleLoadData(goCtx goctx.Context, loadDataInfo *executor
 	return errors.Trace(txn.Commit(goCtx))
 }
 
+// handleLoadStats does the additional work after processing the 'load stats' query.
+// It sends client a file path, then reads the file content from client, loads it into the storage.
+func (cc *clientConn) handleLoadStats(goCtx goctx.Context, loadStatsInfo *executor.LoadStatsInfo) error {
+	// If the server handles the load data request, the client has to set the ClientLocalFiles capability.
+	if cc.capability&mysql.ClientLocalFiles == 0 {
+		return errNotAllowedCommand
+	}
+	if loadStatsInfo == nil {
+		return errors.New("Load stats: info is empty")
+	}
+	err := cc.writeReq(loadStatsInfo.Path)
+	if err != nil {
+		return errors.Trace(err)
+	}
+	var prevData, curData []byte
+	for {
+		curData, err = cc.readPacket()
+		if err != nil {
+			if terror.ErrorNotEqual(err, io.EOF) {
+				log.Error(errors.ErrorStack(err))
+				return errors.Trace(err)
+			}
+		}
+		if len(curData) == 0 {
+			break
+		}
+		prevData = append(prevData, curData...)
+	}
+	if len(prevData) != 0 {
```

Comment:
It seems we shoudn't check this. The `unmarshal` function will handle this case.

Reply:
I think to check it is better, because if you load an non-exist file, this will throw error like `ERROR 2 (HY000): File 'xx' not found (Errcode: 2 "No such file or directory")`, but not `"load stats 'xx'" unexpected end of JSON input`.

Reply:
If the file doesn't exist, will it report error early in line 817 or 811 ?

Reply:
It will not.